VERSION ""

NS_ :
    NS_DESC_
    CM_
    BA_DEF_
    BA_
    VAL_
    CAT_DEF_
    CAT_
    FILTER
    BA_DEF_DEF_
    EV_DATA_
    ENVVAR_DATA_
    SGTYPE_
    SGTYPE_VAL_
    BA_DEF_SGTYPE_
    BA_SGTYPE_
    SIG_TYPE_REF_
    VAL_TABLE_
    SIG_GROUP_
    SIG_VALTYPE_
    SIGTYPE_VALTYPE_
    BO_TX_BU_
    BA_DEF_REL_
    BA_REL_
    BA_DEF_DEF_REL_
    BU_SG_REL_
    BU_EV_REL_
    BU_BO_REL_
    SG_MUL_VAL_

BS_:
BU_: Dashboard PEI TelemNode

BO_ 1894 Vehicle_State: 7 Dashboard
   SG_ Dashboard_HV_Requested : 7|8@0+ (1,0) [0|1] "" Vector__XXX
   SG_ Dashboard_Throttle1_Level : 15|8@0+ (1,0) [0|100] "%" Vector__XXX
   SG_ Dashboard_Throttle2_Level : 23|8@0+ (1,0) [0|100] "%" Vector__XXX
   SG_ Dashboard_Brake_Level : 31|8@0+ (1,0) [0|100] "%" Vector__XXX
   SG_ Dashboard_State : 39|8@0+ (1,0) [0|255] "" Vector__XXX
   SG_ Dashboard_VCU_Ticks : 47|16@0+ (1,0) [0|65535] "ms" Vector__XXX

BO_ 896 BMS_Status: 5 PEI
   SG_ PEI_BMS_Status : 7|8@0+ (1,0) [0|255] "" Vector__XXX
   SG_ PEI_SPI_Error_Flags : 15|16@0+ (1,0) [0|65535] "" Vector__XXX
   SG_ PEI_Max_Faulting_IC_Address : 31|8@0+ (1,0) [0|9] "" Vector__XXX
   SG_ PEI_Communication_Break_ID : 39|8@0- (1,0) [-1|9] "" Vector__XXX

BO_ 903 Status: 5 PEI
   SG_ PEI_IMD_OK : 5|1@0+ (1,0) [0|1] "" Vector__XXX
   SG_ PEI_BMS_OK : 4|1@0+ (1,0) [0|1] "" Vector__XXX
   SG_ PEI_SHUTDOWN_FINAL : 3|1@0+ (1,0) [0|1] "" Vector__XXX
   SG_ PEI_AIR_NEG : 2|1@0+ (1,0) [0|1] "" Vector__XXX
   SG_ PEI_AIR_POS : 1|1@0+ (1,0) [0|1] "" Vector__XXX
   SG_ PEI_PRECHARGE : 0|1@0+ (1,0) [0|1] "" Vector__XXX
   SG_ PEI_Current_ADC : 15|16@0+ (1,0) [0|65535] "Raw_ADC" Vector__XXX
   SG_ PEI_Current_Reference : 31|16@0- (1,0) [-2048|2047] "Raw_ADC" Vector__XXX

BO_ 1024 Cooling_Loop_Temps: 8 TelemNode
   SG_ TelemNode_Inlet_Water_Temp : 7|16@0- (0.1,0) [-3276.8|3276.7] "C" Vector__XXX
   SG_ TelemNode_Outlet_Water_Temp : 23|16@0- (0.1,0) [-3276.8|3276.7] "C" Vector__XXX
   SG_ TelemNode_Air_Into_Radiator_Temp : 39|16@0- (0.1,0) [-3276.8|3276.7] "C" Vector__XXX
   SG_ TelemNode_Air_Out_Of_Radiator_Temp : 55|16@0- (0.1,0) [-3276.8|3276.7] "C" Vector__XXX

BO_ 1026 Cool_Loop_Pressures: 4 TelemNode
   SG_ TelemNode_Inlet_Water_Pressure : 7|16@0+ (0.01,0) [0|1] "PSI" Vector__XXX
   SG_ TelemNode_Outlet_Water_Pressure : 23|16@0+ (0.01,0) [0|1] "PSI" Vector__XXX

BO_ 1025 Wheel_Speed_Rear: 4 TelemNode
   SG_ TelemNode_Wheel_Speed_Rear_Right : 7|16@0+ (1,0) [0|65535] "CPS/RPM" Vector__XXX
   SG_ TelemNode_Wheel_Speed_Rear_Left : 23|16@0+ (1,0) [0|65535] "CPS/RPM" Vector__XXX

BO_ 192 Torque_Request: 8 Dashboard
   SG_ Dashboard_Torque : 0|16@1- (0.1,0) [-3276.8|3276.7] "Nm" Vector__XXX
   SG_ Dashboard_Speed : 16|16@1+ (1,0) [0|65535] "RPM" Vector__XXX
   SG_ Dashboard_Direction : 32|8@1+ (1,0) [0|1] "" Vector__XXX
   SG_ Dashboard_Inverter_Enable : 40|1@1+ (1,0) [0|1] "" Vector__XXX
   SG_ Dashboard_Discharge_Enable : 41|1@1+ (1,0) [0|1] "" Vector__XXX
   SG_ Dashboard_Speed_Mode_Enable : 42|1@1+ (1,0) [0|1] "" Vector__XXX
   SG_ Dashboard_Torque_Limit : 48|16@1+ (0.1,0) [0|6553.5] "Nm" Vector__XXX

BO_ 1280 Random_Shit: 6 Dashboard
   SG_ Dashboard_Front_Strain_Gauge : 7|16@0+ (1,0) [0|65535] "" Vector__XXX
   SG_ Dashboard_Front_Wheel_Speed : 23|16@0+ (1,0) [0|65535] "RPM" Vector__XXX
   SG_ Dashboard_TC_Torque_Request : 39|16@0+ (0.1,0) [0|6553.5] "Nm" Vector__XXX

BO_ 897 Diagnostic_BMS_Data: 4 PEI
   SG_ PEI_HI_Temp : 7|8@0+ (1,0) [0|255] "C" Vector__XXX
   SG_ PEI_SOC : 15|8@0+ (1,0) [0|100] "%" Vector__XXX
   SG_ PEI_Pack_Voltage : 23|16@0- (1,0) [-32768|32767] "V" Vector__XXX

BO_ 1027 Strain_Gauges_Rear: 8 TelemNode
   SG_ TelemNode_RL_Toe_Strain_Gauge : 7|16@0+ (1,0) [0|65535] "" Vector__XXX
   SG_ TelemNode_RLUF_A_Arm_Strain_Gauge : 23|16@0+ (1,0) [0|65535] "" Vector__XXX
   SG_ TelemNode_RLUB_A_Arm_Strain_Gauge : 39|16@0+ (1,0) [0|65535] "" Vector__XXX
   SG_ TelemNode_RLLF_A_Arm_Strain_Gauge : 55|16@0+ (1,0) [0|65535] "" Vector__XXX

BO_ 1281 Inputs: 3 Dashboard
   SG_ Dashboard_Knob1 : 7|8@0+ (1,0) [0|100] "" Vector__XXX
   SG_ Dashboard_Knob2 : 15|8@0+ (1,0) [0|100] "" Vector__XXX
   SG_ Dashboard_DISPLAY_MODE : 20|2@0+ (1,0) [0|1] "" Vector__XXX
   SG_ Dashboard_TC : 18|1@0+ (1,0) [0|1] "" Vector__XXX
   SG_ Dashboard_MARKER : 17|1@0+ (1,0) [0|1] "" Vector__XXX
   SG_ Dashboard_OVERTAKE : 16|1@0+ (1,0) [0|1] "" Vector__XXX

BO_ 904 Current: 4 PEI
   SG_ PEI_Current : 7|16@0- (0.01,0) [-327.68|327.67] "A" Vector__XXX

BO_ 898 High_Level_BMS_Data: 4 PEI
   SG_ PEI_HI_Temp : 7|16@0+ (0.001,0) [0|65.535] "C" Vector__XXX
   SG_ PEI_Pack_Voltage : 23|16@0+ (0.01,0) [0|655.35] "V" Vector__XXX

BO_ 899 BMS_Voltages: 8 PEI
   SG_ PEI_Subpack : 7|8@0+ (1,0) [0|4] "" Vector__XXX
   SG_ PEI_Group : 15|8@0+ (1,0) [0|7] "" Vector__XXX
   SG_ PEI_Voltage_1 : 23|16@0+ (0.01,0) [0|655.35] "V" Vector__XXX
   SG_ PEI_Voltage_2 : 39|16@0+ (0.01,0) [0|655.35] "V" Vector__XXX
   SG_ PEI_Voltage_3 : 55|16@0+ (0.01,0) [0|655.35] "V" Vector__XXX

BO_ 900 BMS_Temps: 8 PEI
   SG_ PEI_Subpack : 7|8@0+ (1,0) [0|4] "" Vector__XXX
   SG_ PEI_Group : 15|8@0+ (1,0) [0|7] "" Vector__XXX
   SG_ PEI_Temp_1 : 23|16@0+ (0.001,0) [0|65.535] "C" Vector__XXX
   SG_ PEI_Temp_2 : 39|16@0+ (0.001,0) [0|65.535] "C" Vector__XXX
   SG_ PEI_Temp_3 : 55|16@0+ (0.001,0) [0|65.535] "C" Vector__XXX

BO_ 1028 Shock_Angles_Rear: 4 TelemNode
   SG_ TelemNode_RL_Shock_Angle : 7|16@0+ (1,0) [0|4095] "Raw_ADC" Vector__XXX
   SG_ TelemNode_RR_Shock_Angle : 23|16@0+ (1,0) [0|4095] "Raw_ADC" Vector__XXX

BO_ 1282 Raw_Throttle: 4 Dashboard
   SG_ Dashboard_Throttle_1 : 7|16@0+ (1,0) [0|65535] "" Vector__XXX
   SG_ Dashboard_Throttle_2 : 23|16@0+ (1,0) [0|65535] "" Vector__XXX

BA_DEF_ BO_ "GenMsgBackgroundColor" STRING ;
BA_DEF_ BO_ "GenMsgForegroundColor" STRING ;
BA_DEF_ BO_ "matchingcriteria" INT 0 0;
BA_DEF_ BO_ "filterlabeling" INT 0 0;
BA_DEF_ BO_ "GenMsgCycleTime" INT 0 100000 ;
BA_DEF_ "BusType" STRING ;

BA_ "GenMsgForegroundColor" BO_ 1894 "#000000";
BA_ "GenMsgForegroundColor" BO_ 896 "#000000";
BA_ "GenMsgForegroundColor" BO_ 903 "#000000";
BA_ "GenMsgForegroundColor" BO_ 1024 "#000000";
BA_ "GenMsgForegroundColor" BO_ 1026 "#000000";
BA_ "GenMsgForegroundColor" BO_ 1281 "#000000";
BA_ "GenMsgForegroundColor" BO_ 899 "#000000";
BA_ "GenMsgForegroundColor" BO_ 900 "#000000";
BA_ "GenMsgForegroundColor" BO_ 1028 "#000000";
BA_ "GenMsgForegroundColor" BO_ 1282 "#000000";

BA_DEF_DEF_ "GenMsgBackgroundColor" "#ffffff";
BA_DEF_DEF_ "GenMsgForegroundColor" "#000000";
BA_DEF_DEF_ "matchingcriteria" 0;
BA_DEF_DEF_ "filterlabeling" 0;
BA_DEF_DEF_ "GenMsgCycleTime" 0;
BA_DEF_DEF_ "BusType" "CAN";

VAL_ 1894 Dashboard_State
    0 "LV"
    1 "PRECHARGE"
    2 "HV ENABLED"
    3 "DRIVE"
    5 "STARTUP"
    129 "DRV FRM LV"
    130 "PRE TM OUT"
    131 "BR NOT PRS"
    132 "HV OFF DRV"
    133 "SNSR DSCRP"
    134 "BSPD TRIPD"
    135 "SHTDWN OPN"
    136 "UNCALIBRTD"
    137 "HARD BSPD"
    138 "MC FAULT";

VAL_ 896 PEI_BMS_Status
    0 "NO ERROR"
    1 "CHARGE MODE"
    2 "BMS TEMP OVER"
    4 "BMS TEMP UNDER"
    8 "OVERVOLT"
    16 "UNDERVOLT"
    32 "OPEN WIRE"
    64 "MISMATCH"
    128 "SPI FAULT";

VAL_ 899 PEI_Group
    0 "A"
    1 "B"
    2 "C"
    3 "D"
    4 "E"
    5 "F"
    6 "G"
    7 "H";